fix: Fix Orama search not working on /docs

smith558 opened the pull request:
## Description

  - **Issue**: Orama search returned no results on `/docs` because the Orama Docusaurus plugin passed the Docusaurus v3 version object straight into `searchParams.where.version`, which expects a string, so the filter never matched.
  - **Cause**: a bug in `@orama/plugin-docusaurus-v3`, it assumes `useVersions()` returns a string; Docusaurus v3 returns version objects.
  - **Fix**: Applied a small `patch-package` diff to coerce the version object to its `.name` (with sensible fallbacks) before building the search filter. Added `patch-package` + `postinstall` to keep the patch applied, documented it in `patches/README.md`, and removed the previous local SearchBar swizzle (initial fix version).

  ### Notes
<img width="1533" height="995" alt="image" src="https://github.com/user-attachments/assets/4bfbf657-8f39-4e88-900d-b27632ee47ac" />

  ## Related Issues

  #348
  https://github.com/fastify/fastify/issues/6390

  ## Check List

  - [x] I have read the Contributing Guidelines (./CONTRIBUTING.md)

jsumners reviewed (approved):
Looks good to me.

mcollina reviewed (commented):
lgtm

mcollina commented:
The test are failing, there seem to be a problem with the image

endymion1818 commented:
@mcollina Oh, you need a `viewBox` attribute, it's stripped by default in SVGO, ok I'll update the PR.

smith558 committed:
Fix Orama search version filter on docs

smith558 committed:
Format SearchBar override

smith558 committed:
Add @orama/react-components dependency

smith558 committed:
Resolve Orama react components import

smith558 commented:
I don't understand why the CI/CD fails. It works locally.

```js
Error:  Error: Unable to build website for locale en.
    at tryToBuildLocale (/home/runner/work/website/website/node_modules/@docusaurus/core/lib/commands/build/build.js:83:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async /home/runner/work/website/website/node_modules/@docusaurus/core/lib/commands/build/build.js:35:9
    at async mapAsyncSequential (/home/runner/work/website/website/node_modules/@docusaurus/utils/lib/jsUtils.js:21:24)
    at async Command.build (/home/runner/work/website/website/node_modules/@docusaurus/core/lib/commands/build/build.js:34:5)
    at async Promise.all (index 0)
    at async runCLI (/home/runner/work/website/website/node_modules/@docusaurus/core/lib/commands/cli.js:56:5)
    at async file:///home/runner/work/website/website/node_modules/@docusaurus/core/bin/docusaurus.mjs:44:3 {
  [cause]: Error: Failed to compile due to Webpack errors.
  Module not found: Error: Can't resolve '@orama/react-components' in '/home/runner/work/website/website/src/theme/SearchBar'
      at /home/runner/work/website/website/node_modules/@docusaurus/bundler/lib/compiler.js:49:24
      at /home/runner/work/website/website/node_modules/webpack/lib/MultiCompiler.js:629:14
      at processQueueWorker (/home/runner/work/website/website/node_modules/webpack/lib/MultiCompiler.js:566:6)
      at process.processTicksAndRejections (node:internal/process/task_queues:85:11)
}
[INFO] Docusaurus version: 3.9.2
Node version: v22.21.1
Error: Process completed with exit code 1.
```

smith558 committed:
Use direct path for Orama react components

smith558 committed:
Use static path for Orama react components

smith558 committed:
Revert "Use static path for Orama react components"

This reverts commit 993e37ff893e6759a8049bfa3071e05461475169.

Eomm commented on src/theme/SearchBar/index.js:
```diff
@@ -0,0 +1,127 @@
+import useOrama from '@orama/plugin-docusaurus-v3/dist/theme/SearchBar/useOrama.js'
+import { getColorMode, getPreferredVersion } from '@orama/plugin-docusaurus-v3/dist/theme/SearchBar/utils.js'
+
+const oramaComponentsPath = '../../node_modules/@orama/react-components/dist/index.js'
```
Why we can't `import @orama/react-components`?

Right now the page does not work:

<img width="844" height="291" alt="Image" src="https://github.com/user-attachments/assets/f63d9b04-020d-48bd-a0b8-77bff4cb0c41" />

smith558 replied:
Yes, that's something I do not understand. It builds and works locally for me just fine. But crashes on GH CI/CD if `import @orama/react-components` used. I was hoping you'd know how to resolve this.

Eomm replied:
Does
`import { OramaSearchButton, OramaSearchBox } from '@orama/react-components'`

work?

Eomm reviewed (commented):
Do we know the offending issue?
Why do we need a wrapper now?

smith558 commented:
> Do we know the offending issue? Why do we need a wrapper now?

The issue is that the Orama plugin assumes `useVersions()` (from `@docusaurus/plugin-content-docs/client`) returns a string, but in Docusaurus v3, it returns version objects. It forwards that object into `searchParams.where.version` (in `OramaSearchBox`), and therefore the query never matches. This is an Orama bug.

smith558 committed:
Patch fix Orama search with Docusaurus v3

smith558 commented:
@Eomm I've reworked this. This fix is much better now. Tested locally as much as possible.

smith558 committed:
Update orama patch for version name handling

smith558 committed:
Ensure patch-package runs during build

smith558 committed:
Update config and lockfile

Eomm commented on patches/@orama+plugin-docusaurus-v3+3.1.16.patch:
```diff
@@ -0,0 +1,31 @@
+diff --git a/node_modules/@orama/plugin-docusaurus-v3/dist/theme/SearchBar/index.js b/node_modules/@orama/plugin-docusaurus-v3/dist/theme/SearchBar/index.js
+index eab29fc..2555aa0 100644
+--- a/node_modules/@orama/plugin-docusaurus-v3/dist/theme/SearchBar/index.js
++++ b/node_modules/@orama/plugin-docusaurus-v3/dist/theme/SearchBar/index.js
```
We should open a PR on their repo instead:
https://github.com/oramasearch/orama/tree/main/packages/plugin-docusaurus-v3

ilteoood replied:
Totally agree with this.
I'll open a PR to Orama to make it happen.

Eomm commented:
> @Eomm I've reworked this. This fix is much better now. Tested locally as much as possible.

Yeah, it looks much better, but if the issue is not in this repo, we should fix this issue for everyone. WDYT?

smith558 commented:
> > @Eomm I've reworked this. This fix is much better now. Tested locally as much as possible.
> 
> Yeah, it looks much better, but if the issue is not in this repo, we should fix this issue for everyone. WDYT?

Yes, I am working on a PR for the upstream as well. However, with the holidays coming, I don't think it will be sooner than a month from now before it gets merged (I am off for 3 weeks starting tomorrow + Orama PR review).

Why not patch it first? I'd put search not working on the docs for at least 3 months as high impact. I am happy to remove the patch later with a PR.

ilteoood commented:
I'll talk to the Orama team to address it ASAP.
A PR with your patch has been opened [here](https://github.com/oramasearch/orama/pull/1005/files)

smith558 commented:
I just saw you got it merged. @ilteoood

ilteoood commented:
You can keep this PR open IMHO, @smith558 
We still need to upgrade the lib once it gets released.
Additionally, do we still need to update docuraurus' config to `lastVersion: 'latest', // the last version is always "latest"`?

smith558 commented:
To be completely frank. I do feel a bit uncomfortable about how this played out.

I spent quite a lot of time debugging this and identifying the root cause, and I was in the process of opening a PR to `oramasearch/orama`. I was just waiting for guidance from @Eomm on the best way to proceed.

This was shaping up to be my first meaningful contribution either to Fastify or Orama, so seeing the same fix copied 1:1 and merged upstream was a bit disappointing.

I’m genuinely glad the issue is fixed for everyone. I just wanted to share how this landed from my side.

ilteoood commented:
I'm sorry about this, but considered your concerns about the time needed for the upstream merge + the fact that you will be on holiday for three weeks, made me realize that probably things needed to be pushed a bit in order to be smoother.
You have also been credited in the PR that has been opened: I've both tagged you and referenced this discussion for historical reasons so...to me, it has been your first meaningful contribution to both the projects.

smith558 commented:
for lib upgrade

Eomm commented:
Nice job @smith558 - Could you cleanup this PR to update the orama plugin required version?

smith558 committed:
Revert previous changes

smith558 committed:
Updates @orama/plugin-docusaurus-v3 to 3.1.18

Updates the @orama/plugin-docusaurus-v3 dependency to the latest version, 3.1.18.

smith558 commented:
this is ready now

ilteoood commented:
Thank you so much for your contribution, @smith558
